test/unit/__snapshots__/switch.spec.ts.snap

```diff
@@ -1,53 +1,66 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`switch not allowed in 5.1: code 1`] = `
-"local ____exports = {}
+exports[`switch produces optimal output 1`] = `
+"require(\\"lualib_bundle\\");
+local ____exports = {}
 function ____exports.__main(self)
-    local ____switch3 = \\"abc\\"
-    goto ____switch3_end
-    ::____switch3_end::
-end
-return ____exports"
-`;
-
-exports[`switch not allowed in 5.1: diagnostics 1`] = `"main.ts(2,9): error TSTL: Switch statements is/are not supported for target Lua 5.1."`;
-
-exports[`switch uses elseif 1`] = `
-"local ____exports = {}
-function ____exports.__main(self)
-    local result = -1
-    local ____switch3 = 2
-    if ____switch3 == 0 then
-        goto ____switch3_case_0
-    elseif ____switch3 == 1 then
-        goto ____switch3_case_1
-    elseif ____switch3 == 2 then
-        goto ____switch3_case_2
-    end
-    goto ____switch3_end
-    ::____switch3_case_0::
-    do
-        do
-            result = 200
-            goto ____switch3_end
-        end
-    end
-    ::____switch3_case_1::
-    do
-        do
-            result = 100
-            goto ____switch3_end
-        end
-    end
-    ::____switch3_case_2::
-    do
-        do
-            result = 1
-            goto ____switch3_end
-        end
-    end
-    ::____switch3_end::
-    return result
+    local out = {}
+    repeat
+        local ____switch3 = 0
+        local ____cond3 = ((____switch3 == 0) or (____switch3 == 1)) or (____switch3 == 2)
+        if ____cond3 then
+            __TS__ArrayPush(out, \\"0,1,2\\")
+        end
+        if ____cond3 then
```

Comment:
The case for the default clause when it is fallen into will always be the condition for the previous condition. There is no additional condition that is needed to be satisfied. In the case above if you passed in any number > 2 it would not satisfy the condition of fallthrough into default and the `not ____cond` default at the bottom is would be triggered.

The `not ___cond` clause is the execution path for default unless there is a situation where default can be fallen into.

Rather than